api_tests/users/views/test_user_claim.py

```diff
@@ -148,7 +148,7 @@ def test_claim_unauth_success_with_unknown_email(self, app, url, project, unreg_
     def test_claim_unauth_success_with_preprint_id(self, app, url, preprint, unreg_user, mock_mail):
         res = app.post_json_api(
             url.format(unreg_user._id),
-            self.payload(email='david@david.son', id=preprint._id),
+            self.payload(email='david@david.son', id=preprint.get_guid()._id),
```

Comment:
hmm, `get_guid()._id` returns the latest version while `id=preprint._id` returns the current version, with this if, I think we are sending the the wrong email to pass the tests.

Here is the failure and I think we may have an actual bug.

```
 FAILED api_tests/users/views/test_user_claim.py::TestClaimUser::test_claim_unauth_success_with_preprint_id - webtest.app.AppError: Bad response: 404 Not Found (not 200 OK or 3xx redirect for http://localhost/v2/users/f8upx/claim/)
{"errors":[{"detail":"Unable to find specified record."}],"meta":{"version":"2.0"}}
```

I will revert the change in this PR and create another ticket on fixing this correctly.